File: 0876-middle-of-the-linked-list/0876-middle-of-the-linked-list.cpp
class Solution {
public:
    ListNode* middleNode(ListNode* head) {
        ListNode* fst = head;
        ListNode* slw = head;
        if(head==NULL) return NULL;
        
        else{
        while(fst!=NULL and fst->next!=NULL)
        {
            fst=fst->next->next;
             slw=slw->next;
        }
           
            return slw;
    }
    }
};

File: 0876-middle-of-the-linked-list/0876-middle-of-the-linked-list.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def middleNode(self, head: ListNode) -> ListNode:
        fst = head
        slw = head
        if not head:
            return None
        else:
            while fst and fst.next:
                fst = fst.next.next
                slw = slw.next
            return slw
